node/src/bootstrapper.rs

```diff
@@ -61,7 +61,7 @@ use tokio::prelude::Stream;
 static mut MAIN_CRYPTDE_BOX_OPT: Option<Box<dyn CryptDE>> = None;
 static mut ALIAS_CRYPTDE_BOX_OPT: Option<Box<dyn CryptDE>> = None;
 
-fn main_cryptde_ref<'a>() -> &'a dyn CryptDE {
+pub fn main_cryptde_ref<'a>() -> &'a dyn CryptDE {
```

Comment:
My IDE isn't very good at finding usages, but when I look for `fn (main|alias)_cryptde`, I find a bunch of functions scattered all over the codebase, some intended to return `CryptDEReal`, some intended to return `CryptDENull`, and some intended to return whichever is appropriate depending on circumstances.

The `CryptDENull` versions do whatever they like, but the `CryptDEReal` versions seem to refer back to here, which is probably why this went from private to public. (I think before this was public, we were relying on the bootstrapper passing the `CryptDEReal`s down into all the places they needed to go.)

Reply:
First, my comment wasn't well written, my apologies. 

However, yeah, I suppose the difference with these functions is that they return a trait object instead of a concrete type. Maybe we could incorporate into the name? If you think it wouldn't be any help then feel free to resolve this conversation right away.